src/librustc_resolve/lib.rs

```diff
@@ -3000,7 +3020,14 @@ impl<'a> Resolver<'a> {
                                             SuggestionType::NotFound => "".to_string(),
                                         }
                                     }
-                                    Field => format!("`self.{}`", path_name),
+                                    Field => {
+                                        is_field = true;
+                                        if self.is_static_method == false {
+                                            format!("`self.{}`", path_name)
+                                        } else {
+                                            "".to_string()
+                                        }
+                                    }
```

Comment:
Just updating the help message to include non-static methods as well would solve it I guess.